src/CaptureClient/ApiEventProcessor.cpp

```diff
@@ -27,132 +27,192 @@ inline std::string DecodeString(const Source& encoded_source) {
+
+  timer_info.set_address_in_function(0);
+
   capture_listener_->OnTimer(timer_info);
 
   asynchronous_events_by_id_.erase(event_id);
 }
 
-void ApiEventProcessor::ProcessTrackingEvent(const orbit_api::ApiEvent& api_event) {
-  TimerInfo timer_info =
-      TimerInfoFromEncodedEvent(api_event.encoded_event, api_event.timestamp_ns,
-                                api_event.timestamp_ns, api_event.pid, api_event.tid, /*depth=*/0);
-  capture_listener_->OnTimer(timer_info);
+void ApiEventProcessor::ProcessStringEventLegacy(const orbit_api::ApiEvent& api_event) {
+  ApiStringEvent api_string_event;
+  api_string_event.set_process_id(api_event.pid);
+  api_string_event.set_thread_id(api_event.tid);
+  api_string_event.set_timestamp_ns(api_event.timestamp_ns);
+  api_string_event.set_async_scope_id(api_event.encoded_event.event.data);
+  api_string_event.set_name(api_event.encoded_event.event.name);
+  capture_listener_->OnApiStringEvent(api_string_event);
 }
 
-namespace {
-void EncodedColorToColor(uint32_t encoded_color, orbit_client_protos::Color* color) {
-  color->set_red((encoded_color & (0xff << 24)) >> 24);
-  color->set_green((encoded_color & (0xff << 16)) >> 16);
-  color->set_blue((encoded_color & (0xff << 8)) >> 8);
-  color->set_alpha(encoded_color & 0xff);
+void ApiEventProcessor::ProcessTrackingEventLegacy(const orbit_api::ApiEvent& api_event) {
+  ApiTrackValue api_track_value;
+  api_track_value.set_process_id(api_event.pid);
+  api_track_value.set_thread_id(api_event.tid);
+  api_track_value.set_timestamp_ns(api_event.timestamp_ns);
+
+  api_track_value.set_name(api_event.encoded_event.event.name);
+
+  switch (api_event.Type()) {
+    case orbit_api::kTrackInt:
+      api_track_value.set_data_int(orbit_api::Decode<int32_t>(api_event.encoded_event.event.data));
+      break;
+    case orbit_api::kTrackInt64:
+      api_track_value.set_data_int64(
+          orbit_api::Decode<int64_t>(api_event.encoded_event.event.data));
+      break;
+    case orbit_api::kTrackUint:
+      api_track_value.set_data_uint(
+          orbit_api::Decode<uint32_t>(api_event.encoded_event.event.data));
+      break;
+    case orbit_api::kTrackUint64:
+      api_track_value.set_data_uint64(
+          orbit_api::Decode<uint64_t>(api_event.encoded_event.event.data));
+      break;
+    case orbit_api::kTrackFloat:
+      api_track_value.set_data_float(orbit_api::Decode<float>(api_event.encoded_event.event.data));
+      break;
+    case orbit_api::kTrackDouble:
+      api_track_value.set_data_double(
+          orbit_api::Decode<double>(api_event.encoded_event.event.data));
+      break;
+
+    case orbit_api::kNone:
+      [[fallthrough]];
+    case orbit_api::kScopeStart:
+      [[fallthrough]];
+    case orbit_api::kScopeStop:
+      [[fallthrough]];
+    case orbit_api::kScopeStartAsync:
+      [[fallthrough]];
+    case orbit_api::kScopeStopAsync:
+      [[fallthrough]];
+    case orbit_api::kString:
```

Comment:
nit. fallthrough is not required here, see go/cstyle#Loops_and_Switch_Statements.

Reply:
But yeah, I know it's code that was copied...